_ingest-pipelines/processors/dissect.md

```diff
@@ -0,0 +1,526 @@
+        "append_separator": "|"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+You can test the pipeline using the following example:
+
+```json
+POST _ingest/pipeline/dissect-test/_simulate
+{
+  "docs": [
+    {
+      "_index": "testindex1",
+      "_id": "1",
+      "_source": {
+        "message": "New York, NY 10017"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+The substrings are appended into the `address` field in reverse order, as shown in the following response:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "testindex1",
+        "_id": "1",
+        "_source": {
+          "address": "10017|NY|New York",
+          "message": "New York, NY 10017"
+        },
+        "_ingest": {
+          "timestamp": "2024-01-22T22:38:24.305974178Z"
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Named skip key
+
+The named skip key modifier excludes specific matches from the final output by using an empty key `{}` or `?` modifier within the pattern. For example, the following patterns are equivalent: `%{firstName} %{lastName} %{?ignore}` and `%{firstName} %{lastName} %{}`. The named skip key modifier is useful for excluding irrelevant or unnecessary fields from the output.
+
+#### Example of usage
+
+The following pattern uses a named skip key to exclude a field (in this case, `ignore`) from the output. You can assign a descriptive name to the empty key, for example, `%{?ignore}`, to clarify that the corresponding value should be excluded from the final result:
+
+```json
+PUT /_ingest/pipeline/dissect-test
+{
+  "description": "Pipeline that dissects web server logs",
+  "processors": [
+    {
+      "dissect": {
+        "field": "message",
+        "pattern": "%{firstName} %{lastName} %{?ignore}"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+You can test the pipeline using the following example:
+
+```json
+POST _ingest/pipeline/dissect-test/_simulate
+{
+  "docs": [
+    {
+      "_index": "testindex1",
+      "_id": "1",
+      "_source": {
+        "message": "John Doe M.D."
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+Your response should be similar to the following:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "testindex1",
+        "_id": "1",
+        "_source": {
+          "firstName": "John",
+          "lastName": "Doe",
+          "message": "John Doe M.D."
+        },
+        "_ingest": {
+          "timestamp": "2024-01-22T22:41:58.161475555Z"
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Reference keys (`*` and `&`)
+
+Reference keys use parsed values as key/value pairings for structured content. This can use useful when handling systems that partially log data in key/value pairs. By using reference keys, you can preserve the key/value relationship and maintain the integrity of the extracted information. 
```

Comment:
```suggestion
Reference keys use parsed values as key-value pairings for structured content. This can use useful when handling systems that partially log data in key-value pairs. By using reference keys, you can preserve the key-value relationship and maintain the integrity of the extracted information. 
```